server/forge/github/github.go

```diff
@@ -667,21 +669,20 @@ func (c *client) loadChangedFilesFromPullRequest(ctx context.Context, pull *gith
 	return pipeline, err
 }
 
-func (c *client) getTagCommitSHA(ctx context.Context, repo *model.Repo, tagName string) (string, error) {
+func (c *client) getTagCommitSHA(ctx context.Context, repo *model.Repo, tagName string) (*model.Commit, error) {
```

Comment:
Similar to gitea / forgejo I would improve those two helpers to do less "magic" and rather one thing like one returning the sha and one the commit from a sha or similar.

```suggestion
func (c *client) getCommitAndMessageFromTag(ctx context.Context, repo *model.Repo, tagName string) (*model.Commit, error) {
```